lib/ansible/modules/files/jsonmod.py

```diff
@@ -0,0 +1,370 @@
+    required: false
+    default: "no"
+    choices: ["yes", "no"]
+  data_json:
+    description:
+      - A valid JSON serialized string. This will be decoded and placed at
+        the path in the JSON document specified by I(json_path).
+    required: false
+    default: null
+  data_str:
+    description:
+      - A string to be placed at the path in the JSON document specified by
+        I(json_path). This will not be decoded as JSON.
+    required: false
+    default: ''
+  json_path:
+    description:
+      - The path to a point inside the JSON document that should be updated.
+    required: true
+    default: null
+  path:
+    description:
+      - Path to a file containing a JSON document.
+    required: true
+    default: null
+  remove:
+    description:
+      - Instead of adding or replacing the node at the path specified by
+        I(json_path), ensure it is removed.
+    required: false
+    default: "no"
+    choices: ["yes", "no"]
+'''
+
+EXAMPLES = '''
+# Before:
+# {"param1": [1, 2], "param2": [3, 4]}
+- name: Replace param1 values
+  jsonmod:
+    data_json: [5, 6]
+    json_path: "param1"
+    path: "/path/to/my.json"
+# After:
+# {"param1": [5, 6], "param2": [3, 4]}
+
+# Before:
+# {"droids": [{"name": "C-4PO"}, {"name": "R2D2"}]}
+- name: Fix typo in droid name
+  jsonmod:
+    data_str: "C-3PO"
+    json_path: "droids[0]name"
+    path: "/path/to/droids.json"
+# After:
+# {"droids": [{"name": "C-3PO"}, {"name": "R2D2"}]}
+
+# Before:
+# {"droids": [{"name": "C-3PO"}, {"name": "R2D2"}]}
+- name: Add droid to list
+  jsonmod:
+    data_json: {"name": "BB-8"}
+    json_path: "droids[2]"
+    path: "/path/to/droids.json"
+# After:
+# {"droids": [{"name": "C-3PO"}, {"name": "R2D2"}, {"name": "BB-8"}]}
+
+# Before:
+# {"train": {"type": "land"}, "aeroplane": {"type": "air", "speed": "fast"}}
+- name: Add speed node to train info
+  jsonmod:
+    data_str: "not as fast"
+    json_path: "train.speed"
+    path: "/path/to/vehicles.json"
+# After:
+# {"train": {"type": "land", "speed": "not as fast"}, "aeroplane": {"type": "air", "speed": "fast"}}
+
+# Before:
+# ["item1", "item2", "item3"]
+- name: Remove item2 from the list
+  jsonmod:
+    json_path: "[1]"
+    path: "/path/to/items.json"
+    remove: yes
+# After:
+# ["item1", "item3"]
+
+# Before:
+# {"item1": {"subitem1": [1, 2], "subitem2": [3, 4]}, "item2": {"subitem4": [7, 8]}}
+- name: Remove subitem3
+  jsonmod:
+    json_path: "item2.subitem3"
+    path: "/path/to/items.json"
+    remove: yes
+# After (notice no change):
+# {"item1": {"subitem1": [1, 2], "subitem2": [3, 4]}, "item2": {"subitem4": [7, 8]}}
+
+# Example where file doesn't currently exist
+- name: Configure feature X
+  jsonmod:
+    create_file: yes
+    data_json: "[1, 2, 3]"
+    json_path: "groupA.featureX"
+    path: "/path/to/configfile.json"
+# After (file now exists):
+# {"groupA": {"featureX": [1, 2, 3]}}
+'''
+
+from collections import OrderedDict
+import copy
+import json
```

Comment:
This will also need to be updated for python 2.4 (thanks @abadger for pointing this out):

```python
try:
    import json
except ImportError:
    import simplejson as json
```